src/cogent3/core/new_alignment.py

```diff
@@ -2288,97 +2322,122 @@ def _(seqs: dict) -> SupportsFeatures:
     return merged_db_collection(seqs.values())
 
 
-@singledispatch
-def coerce_to_seqs_data_dict(
-    data, label_to_name: OptRenamerCallable = None
-) -> dict[str, StrORBytesORArray]:
-    # refactor: handle conversion of SeqView to SeqDataView.
-    raise NotImplementedError(
-        f"coerce_to_seqs_data_dict not implemented for {type(data)}"
-    )
+@dataclasses.dataclass
+class raw_seq_data:
+    seq: StrORBytesORArray
+    name: OptStr = None
+    parent_name: OptStr = None
+    offset: int = 0
+    is_reversed: bool = False
 
 
-@coerce_to_seqs_data_dict.register
-def _(
-    data: dict, label_to_name: OptRenamerCallable = None
-) -> dict[str, StrORBytesORArray]:
-    is_sequence = isinstance(next(iter(data.values()), None), new_sequence.Sequence)
-    return {
-        (label_to_name(k) if label_to_name else k): (
-            numpy.array(v) if is_sequence else v
-        )
-        for k, v in data.items()
-    }
+@singledispatch
+def coerce_to_raw_seq_data(
```

Comment:
what's the reason for using `isinstance` instead of having a dispatch for `Aligned` instances?

Reply:
because Aligned is defined below so that dispatch system won't work because its not defined